Libs/MRML/Core/vtkMRMLModelStorageNode.cxx

```diff
@@ -332,6 +340,12 @@ int vtkMRMLModelStorageNode::WriteDataInternal(vtkMRMLNode *refNode)
     return 0;
   }
 
+  if (modelNode->GetMeshConnection() == nullptr)
```

Comment:
Can you check if we can write a polygon that has 0 points? If not, then it would be probably better to treat it as CancelledNoData, too.